internal/appconfig/config_test.go

```diff
@@ -205,3 +206,35 @@ func TestHasNonHttpAndHttpsStandardServices(t *testing.T) {
 	}}}
 	assert.True(t, cfg6.HasNonHttpAndHttpsStandardServices())
 }
+
+func TestURLCalculation(t *testing.T) {
+	port80 := 80
+	port443 := 443
+
+	http, _ := url.Parse("http://test.fly.dev")
+	https, _ := url.Parse("https://test.fly.dev")
+
+	cfg := NewConfig()
+	cfg.AppName = "test"
+	cfg.Services = []Service{{Protocol: "tcp", Ports: []api.MachinePort{
+		{Port: &port80, Handlers: []string{"http"}},
+	}}}
+	url, _ := cfg.URL()
+	assert.Equal(t, http, url)
+
+	cfg = NewConfig()
+	cfg.AppName = "test"
+	cfg.Services = []Service{{Protocol: "tcp", Ports: []api.MachinePort{
+		{Port: &port443, Handlers: []string{"tls"}},
```

Comment:
"tls" alone doesn't guarantee an http service, the correct handlers are `[]string{"tls", "http"}`